pinot-udf/build.gradle.kts

```diff
@@ -0,0 +1,16 @@
+plugins {
+  `java-library`
+}
+
+// Pinot images run with Java 11, not JDK 14
+//
+// If any other dependencies are added, make sure they also are limited accordingly
+configure<JavaPluginExtension> {
```

Comment:
this was why it was broke

Reply:
@adriancole could you please elaborate on the why so people like me can understand it? 🥛

Reply:
We decided to not use latest java image at runtime (not sure why, as it wasn't my recommendation).

When projects don't set a java compile version (they should always do this), it is implicitly taken from whatever version the compiler (JDK) is. So, this implicitly compiled at Java 14.

Pinot uses a crappy reflections library to auto-load `@ScalarFunction` by scanning the classpath. (Note: this is also why flat classpath is faster bootstrapping vs default which would be unbundling jars). That reflections thing bombs with a generic message when there's any problem loading a class https://github.com/ronmamo/reflections/blob/master/src/main/java/org/reflections/util/Utils.java#L68-L81

What happened here was Java 11 was used to load a Java 14 class, and basically I made a lucky guess!

Reply:
Thanks!

Reply:
The comments I made above, btw, I wouldn't make in any other project as it is more normal to set a lower java library compile level always. Ex most set 1.8 even if they use JRE 14. Since we tripped over this a few times in projects here, I elaborated the comment until folks learn to do this, or a general config does this implicitly.

Reply:
Yeah I actually had this locally and have been holding it (this PR is incomplete and was not meant for review, hence the comment and draft status :) ) to remind myself to resolve the fact that our two flavors of pinot are running on different jdk versions right now - the udfs would actually need to go all the way down to 8, so want to bump pinot to 11 across the board instead. A TODO probably would have been easier.